src/rsqsim_api/rsqsim_api/catalogue/catalogue.py

```diff
@@ -440,8 +443,11 @@ def events_by_number(self, event_number: Union[int, np.int, Iterable[np.int]], f
                                                            patch_time=patch_time_list,
                                                            fault_model=fault_model, min_patches=min_patches,
                                                            event_id=index)
+                if generator:
```

Comment:
What is the generator doing? (sure it's fine but would be good to comment/ add a docstring)